lib/downloader.rb

```diff
@@ -40,6 +40,11 @@ def downloader (*args)
     when 'file'
       # use FileUtils to copy if it is a local file (the url protocol is file://)
       if File.exist?(uri.path)
+        if args[1].to_s.empty?
+          filename = File.basename(uri.path)
+        else
+          filename = args[1]
+        end
         return FileUtils.cp uri.path, filename
```

Comment:
If `args[1]` is a string, we don't need to check if it is valid (easier for debugging, also since this won't involve env variables, it is unlikely that an empty string will be passed instead of `nil` (why pass it if it is empty :) ) )

As above, we can combine the `if else` part with the `return` part:
```ruby
return FileUtils.cp uri.path, args[1] || File.basename(uri.path)
```

Reply:
I'm all for simplifying...if it works. :)